service/worker/processor.go

```diff
@@ -83,7 +85,7 @@ var (
 )
 
 func newReplicationTaskProcessor(currentCluster, sourceCluster, consumer string, client messaging.Client, config *Config,
-	logger bark.Logger, metricsClient metrics.Client, domainReplicator DomainReplicator,
+	logger bark.Logger, metricsClient metrics.Client, domainCache cache.DomainCache, domainReplicator DomainReplicator,
```

Comment:
How about we emit target clusters on the replication task.  This way worker can filter that by just looking at the replication task without needing domainCache.